### ISSUE TITLE
please add some comments on these lines

### ISSUE THREAD

**Issue by yorkerlin** [emtiyaz/vadam#1]
https://github.com/emtiyaz/vadam/blob/c09d3ac2d6ebabf56f69ee4858926a949dbfeab2/pytorch/vadam/optimizers.py#L67

See also: https://github.com/emtiyaz/vadam/blob/c09d3ac2d6ebabf56f69ee4858926a949dbfeab2/pytorch/vadam/optimizers.py#L92

Make sure that the same mini-batch should be used here.
One way to achieve that is to use `state['step']` as the seed of the random number generator for generating mini-batches. I did that in my Autograd implementation

**Comment by yorkerlin** [comment]
It will be nice to point out that the loss function in https://github.com/emtiyaz/vadam/blob/c09d3ac2d6ebabf56f69ee4858926a949dbfeab2/pytorch/vadam/optimizers.py#L92 computes the `mean` (instead of the `sum`) of the loss over data points in one mini-batch.
